### PR TITLE
[ACM-8821]: Adding bare metal doc about adding node pools

### PR DESCRIPTION
**Pull request by lahinson**
Related Jira: https://issues.redhat.com/browse/ACM-8821

### CODE REVIEW

**Comment by avishayt** [comment]
Hi Laura, thanks for creating this!
A few comments:
- The new subsection is under the generic platform-independent section, but is meant only for the agent platform
- The command parameters like --cores, --memory, and --root-volume-size are specific to the kubevirt platform and are not valid for the agent platform
- Unfortunately we didn't implement this for the agent platform yet.  I opened ACM-8917 to address this.

**Comment by lahinson** [comment]
@avishayt Thanks for your feedback! I created this PR at Gal's request, as he was looking for agent-specific docs about adding a node pool. A few responses to your comments:

>The new subsection is under the generic platform-independent section, but is meant only for the agent platform

I could be mistaken, but I thought the file that I added the content to was agent-specific. That file appears only in the bare metal section of the HyperShift docs.

>The command parameters like --cores, --memory, and --root-volume-size are specific to the kubevirt platform and are not valid for the agent platform

Good to know! I will remove those from the PR.

>Unfortunately we didn't implement this for the agent platform yet. I opened ACM-8917 to address this.

Ah, okay. Should I put this PR on hold then? Just for your awareness, my last day of work for the year is 15 December, and then I will be on maternity leave for a few months. If we need to put this on hold, I can let another doc team member know so that someone else can merge this at the right time.

**Comment by avishayt** [comment]
Congrats!  I already posted the PR to implement this and expect it to be merged soon.  I'm not sure what the timing is between the release and the doc update.  Probably not critical if the docs land first as this isn't a very popular feature.

**Comment by lahinson** [comment]
@avishayt Thanks 😄 

Do you think it would be okay if I went ahead and had this PR peer-reviewed, and then I can merge whenever ACM-8917 is merged?

File: clusters/hosted_control_planes/scaling_nodepool_hosted.adoc [hunk 1]
```diff
@@ -114,4 +132,43 @@ clusteroperator.config.openshift.io/operator-lifecycle-manager-catalog         4
 clusteroperator.config.openshift.io/operator-lifecycle-manager-packageserver   4.12z    True        False         False      40m     
 clusteroperator.config.openshift.io/service-ca                                 4.12z    True        False         False      11m     
 clusteroperator.config.openshift.io/storage                                    4.12z    True        False         False      11m 
+----
+
+[#adding-nodepool-bm]
+== Adding node pools
+
+You can create node pools for a hosted cluster by specifying a name, number of replicas, and any additional information, such as memory and CPU requirements.
```
Comment (avishayt) [inline clusters/hosted_control_planes/scaling_nodepool_hosted.adoc]:
```suggestion
You can create node pools for a hosted cluster by specifying a name, number of replicas, and any additional information, such as an agent label selector.
```

File: clusters/hosted_control_planes/scaling_nodepool_hosted.adoc [hunk 2]
```diff
@@ -114,4 +132,43 @@ clusteroperator.config.openshift.io/operator-lifecycle-manager-catalog         4
...
+
+. To create a node pool, enter the following information. In this example, the node pool has more CPUs assigned to the VMs:
```
Comment (avishayt) [inline clusters/hosted_control_planes/scaling_nodepool_hosted.adoc]:
```suggestion
. To create a node pool, enter the following information. In this example, the node pool will use agents with the label "size=medium":
```

File: clusters/hosted_control_planes/scaling_nodepool_hosted.adoc [hunk 3]
```diff
@@ -114,4 +132,43 @@ clusteroperator.config.openshift.io/operator-lifecycle-manager-catalog         4
...
+
++
+----
+export NODEPOOL_NAME=${CLUSTER_NAME}-extra-cpu
+export WORKER_COUNT="2"
+
+hcp create nodepool agent \
+  --cluster-name $CLUSTER_NAME \
+  --name $NODEPOOL_NAME \
+  --node-count $WORKER_COUNT \
```
Comment (avishayt) [inline clusters/hosted_control_planes/scaling_nodepool_hosted.adoc]:
```suggestion
hcp create nodepool agent \
  --cluster-name $CLUSTER_NAME \
  --name $NODEPOOL_NAME \
  --node-count $WORKER_COUNT \
  --agentLabelSelector '{"matchLabels": {"size": "medium"}}'
```

**Comment by avishayt** [comment]
> @avishayt Thanks 😄
> 
> Do you think it would be okay if I went ahead and had this PR peer-reviewed, and then I can merge whenever ACM-8917 is merged?

I added a couple comments.  ACM-8917 is merged but not yet released.  I think you can go ahead and merge this, though.

File: clusters/hosted_control_planes/scaling_nodepool_hosted.adoc (hunk 2 above)
Comment (lahinson) [reply clusters/hosted_control_planes/scaling_nodepool_hosted.adoc]: @avishayt In the code, it looks like the label is `{"size": "medium"}`. Should I use that exact wording on line 142 as well?

Comment (avishayt) [reply clusters/hosted_control_planes/scaling_nodepool_hosted.adoc]: Yes, the best way would be like in the Kubernetes docs:
https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/

So `"size" : "medium"`

**Comment by avishayt** [comment]
LGTM thanks!

**Review by dockerymick — APPROVED** [review]
looks good, thanks !

**Pull request merged**